### PR TITLE
Remove hardwired return type in ExecuteScriptCommand

### PR DESCRIPTION
**Pull request by isaacvale**
ExecuteScriptCommand return type should be generic. This PR removes the hardwired return type, pushes it up to Program.cs and passes generic type.

### CODE REVIEW

**Comment by filipw** [comment]
Can you explain your use case?

**Review by seesharper — APPROVED** [review]

File: src/Dotnet.Script.Core/Commands/PublishCommand.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public PublishCommand(ScriptConsole scriptConsole, LogFactory logFactory)
             _logFactory = logFactory;
         }
 
-        public void Execute(PublishCommandOptions options)
+        public void Execute<TReturn>(PublishCommandOptions options)
```
Comment (filipw) [inline src/Dotnet.Script.Core/Commands/PublishCommand.cs:20]: makes sense thanks. It's technically a breaking change - could you please keep the old one too and add the generic as a new overload?

It does not matter that much but I would like to have backwards compatibility in the public API unless we rev up major version

Comment (isaacvale) [reply src/Dotnet.Script.Core/Commands/PublishCommand.cs:20]: Yes, you're right, should be an overload, backwards compatibility is very important. Sure, I'll add it

**Review by filipw — APPROVED** [review]
thanks!

**Pull request merged**